src/gluonts/itertools.py

```diff
@@ -113,3 +122,6 @@ def __init__(self, iterable: Iterable, length: Optional[int]) -> None:
 
     def __iter__(self):
         return itertools.islice(self.iterable, self.length)
+
+    def __len__(self) -> int:
+        return len(self.iterable)
```

Comment:
Is this needed? And what happens if `self.iterable` doesn't support `len`?

Reply:
It's required for setting the max number of idle transformations. If it doesn't support `len`, it'll throw (probably an `AttributeError`). The base datasets always should implement it, right?

Reply:
Yes, but this class doesn't operate on `Dataset` but on `Iterable`.

Reply:
Sure, but the current behavior won't change if `self.iterable` does not provide a `__len__`. If we don't put this here, the entire PR is useless since datasets are most often being transformed :D I just noticed though that we need to catch whether `__len__` throws when setting `max_idle_transforms` ... I will fix that